File: src/backend/SDM/SDM.Domain/Entities/Device.cs
using SDM.Domain;

namespace SDM.Domain.Entities
{
    public class Device
    {
        public Guid Id { get; set; }

        public string DeviceIdentifier { get; set; } = string.Empty; // e.g., Android device id

        public string SerialNumber { get; set; } = string.Empty;

        public string Manufacturer { get; set; } = string.Empty;

        public string Model { get; set; } = string.Empty;

        public string AndroidVersion { get; set; } = string.Empty;

        public int BatteryLevel { get; set; }

        public DateTime? LastSeen { get; set; }

        public DeviceStatus Status { get; set; } = DeviceStatus.Offline;

        public Guid? GroupId { get; set; }

        public string? AssignedUserName { get; set; }

        public ComplianceStatus ComplianceStatus { get; set; } = ComplianceStatus.Unknown;

        public ManagementMode ManagementMode { get; set; } = ManagementMode.CustomAgent;

        // Android Management API device resource name (e.g. "enterprises/LC.../devices/abc123"),
        // set only for AndroidEnterprise* devices. Used for Policy/issueCommand calls in Phase 3.
        public string? GoogleDeviceName { get; set; }

        public DateTime CreatedOn { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedOn { get; set; }

        public ICollection<DeviceCommand> Commands { get; set; } = new List<DeviceCommand>();

        public ICollection<DeviceHeartbeat> Heartbeats { get; set; } = new List<DeviceHeartbeat>();

        public ICollection<DevicePushToken> PushTokens { get; set; } = new List<DevicePushToken>();

        public ICollection<DeviceViolation> Violations { get; set; } = new List<DeviceViolation>();
    }
}

File: src/frontend/src/types/device.ts
export enum DeviceStatus {
  Offline = 0,
  Online = 1,
  Inactive = 2,
}

export enum ComplianceStatus {
  Unknown = 0,
  Compliant = 1,
  NonCompliant = 2,
  Pending = 3,
}

export enum EnrollmentType {
  Corporate = 0,
  BYOD = 1,
}

export enum ManagementMode {
  Unknown = 0,
  DeviceOwner = 1,
  ProfileOwner = 2,
}

export interface Device {
  id: string;
  deviceIdentifier: string;
  serialNumber: string;
  manufacturer: string;
  model: string;
  androidVersion: string;
  batteryLevel: number;
  lastSeen: string | null;
  status: DeviceStatus;
  complianceStatus: ComplianceStatus;
  assignedUserName: string | null;
  groupId: string | null;
  enrollmentType: EnrollmentType;
  managementMode: ManagementMode;
  createdOn: string;
  updatedOn: string | null;
}

export interface DeviceCommand {
  id: string;
  deviceId: string;
  commandType: string;
  payload: string;
  status: number;
  retryCount: number;
  maxRetries: number;
  createdOn: string;
  executedOn: string | null;
  acknowledgedOn: string | null;
  batchId: string | null;
  createdByUserId: string | null;
  createdByName: string | null;
}

export interface DeviceViolation {
  id: string;
  description: string;
  createdOn: string;
}

export interface DeviceGroup {
  id: string;
  name: string;
  description: string | null;
  category: string | null;
  deviceCount: number;
}

export interface PagedResult<T> {
  items: T[];
  total: number;
  page: number;
  pageSize: number;
}
